### ISSUE TITLE
android_alarm_manager doesn't work once you leave the app

### ISSUE THREAD

**Issue by alexandranb** [flutter/flutter#26073]
Hello,

I have some issues with the android_alarm_manager plugin when scheduling periodically to display some local notifications.

1. Firstly, I can't make it run when the app is terminated, and from what I read, this plugin should be able to do that.
2. I don't want to use only the local notifications plugin to schedule the notifications, because I need to show the notifications only if it meets some requirements.
3. When I close the app I get the following error:

```
I/flutter (19593): onSelectLocalNotification 2019-01-04 13:50:36.131706
I/AlarmService(19593): setBackgroundFlutterView tried to overwrite an existing FlutterNativeView
W/art     (19593): Native thread exiting without having called DetachCurrentThread (maybe it's going to use a pthread_key_create destructor?): Thread[16,tid=19649,Native,Thread*=0x99d6a500,peer=0x130360a0,"Thread-1126"]
Lost connection to device.
```

## flutter doctor -v output
```
[√] Flutter (Channel beta, v1.0.0, on Microsoft Windows [Version 6.3.9600], locale en-US)
    • Flutter version 1.0.0 
    • Framework revision 5391447fae (5 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at ...
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = ...
    • Java binary at: ....
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[√] Android Studio (version 3.2)
    • Android Studio at ...
    • Flutter plugin version 31.3.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[√] VS Code, 32-bit edition (version 1.29.1)
    • VS Code at ...
    • Flutter extension version 2.21.1

[√] Connected device (1 available)
    • ONE E1001 • 6698bf2e • android-arm • Android 6.0.1 (API 23)

• No issues found!

```

**Comment by Hixie** [comment]
I also see this. This seems like a major problem since it defeats a large part of the purpose of the plugin.

**Comment by bkonyi** [comment]
This should be resolved by flutter/plugins#1129.
